pageserver/src/layered_repository.rs

```diff
@@ -308,6 +308,35 @@ impl Repository for LayeredRepository {
         Ok(())
     }
 
+    fn materialize_iteration(&self) -> Result<()> {
+        // Scan through the hashmap and collect a list of all the timelines,
+        // while holding the lock. Then drop the lock and actually perform the
+        // reconstructions.  We don't want to block everything else while the
+        // reconstruction runs.
+        let timelines = self.timelines.lock().unwrap();
+        let timelines_to_reconstruct = timelines
+            .iter()
+            .map(|(timelineid, timeline)| (*timelineid, timeline.clone()))
+            .collect::<Vec<_>>();
+        drop(timelines);
+
+        for (timelineid, timeline) in &timelines_to_reconstruct {
+            let _entered =
+                info_span!("reconstruct", timeline = %timelineid, tenant = %self.tenantid)
+                    .entered();
+            match timeline {
+                LayeredTimelineEntry::Loaded(timeline) => {
+                    timeline.reconstruct()?;
+                }
+                LayeredTimelineEntry::Unloaded { .. } => {
+                    debug!("Cannot reconstruct remote timeline {}", timelineid)
```

Comment:
Unloaded timeline is not a remote one. It is available on the local disk but without loaded layermap and other in memory structures

Reply:
Currently, all timelines are loaded during the GC but I think it should be possible to perform GC on an unloaded timeline

Reply:
Sorry, this fragment of code was copied from  `compaction_iteration` above.
So ifmessage/logic is not correct, then it should be also applied to `compaction_iteration`

Reply:
Also with new disk_btree2 there will be no need in unloading layers from memory(i.e. layer index from memory). But still concept of loaded/unloaded layer may be useful for layers swapped to S3. So I think that them can be reused in new sense, replacing notion of remote layer.